pkgs/build-support/dotnet/build-dotnet-module/hooks/dotnet-install-hook.sh

```diff
@@ -7,17 +7,20 @@ dotnetInstallHook() {
     runHook preInstall
 
     if [ "${selfContainedBuild-}" ]; then
-        dotnetInstallFlags+=("--self-contained")
+        dotnetInstallFlags+=(--runtime "@runtimeId@" "--self-contained")
```

Comment:
Why is this only for selfContainedBuild?

Reply:
Sorry for the late reply, but `--self-contained` without '--runtime' results in:

```
/nix/store/0jarb9qww0jg62lw6mlkjgs9fx3gxw98-dotnet-sdk-6.0.300/sdk/6.0.300/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(151,5): error NETSDK1031: It is not supported to build or publish a self-contained application without specifying a RuntimeIdentifier. You must either specify a RuntimeIdentifier or set SelfContained to false. [/home/davidm/tmp/tmp.csproj]
```

`--no-self-contained` works without a runtime id.  Presumably the rid only applies when making a self-contained executable.

Reply:
If `--runtime` is omitted in a `--no-self-contained` build, native libraries for all platforms will be installed in `result/lib/<name>/runtimes`. There is actually an example using `--runtime` and `--self-contained false` together [in dotnet documentations](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-publish#examples).